### PR TITLE
[PAY-925] Adds unread message icons on web

### PR DESCRIPTION
**Pull request by schottra**
### Description
This implements unread message notifications at the navigation level for web.

* Added an `Icon` helper component which can be used to load any of our icons and optionally add a decorator component
* Added `NotificationDot` component with small / large size variants
* Updated nav popup component to render notification dots on the icon and messages menu item when user has unread chat messages

### Dragons
N/A

### How Has This Been Tested?
Manual testing on web

### How will this change be monitored?
N/A

### Feature Flags ###
N/A

image

### CODE REVIEW

**Review by rickyrombo — APPROVED** [review]
_love_ ❤️ 

EDIT: check stylelint in CI

**Comment by schottra** [comment]
> Looks great! Worth flagging this `Icon` component to the client team / design system migration effort at some point

cc @dylanjeffers @sliptype

File: packages/web/src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+/** Renders a stems Icon component with optional decorator
+ * Ex: `<Icon icon={IconKebabHorizontal} decorator={<NotificationDot variant='small' />} />`
+ */
+export const Icon = ({
+  decorator,
+  icon: IconComponent,
+  ...iconProps
+}: IconProps) => {
+  return decorator ? (
+    <div className={styles.iconContainer}>
```
Comment (dylanjeffers) [inline packages/web/src/components/Icon/Icon.tsx:20]: spans maybe?

File: packages/web/src/components/Icon/Icon.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      <IconComponent />
+      <div className={styles.iconDecoration}>{decorator}</div>
+    </div>
+  ) : (
+    React.createElement(IconComponent, iconProps)
```
Comment (dylanjeffers) [inline packages/web/src/components/Icon/Icon.tsx:25]: cant just do <IconComponent {...iconProps} /> ?

File: packages/web/src/components/notification-dot/NotificationDot.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import styles from './NotificationDot.module.css'
+
+type NotificationDotVariant = 'small' | 'large'
+type NotificationDotProps = {
+  variant: NotificationDotVariant
+} & ComponentProps<'span'>
+
+/** Renders a basic notification dot used to decorate controls and indicate new information */
+export const NotificationDot = (props: NotificationDotProps) => {
```
Comment (dylanjeffers) [inline packages/web/src/components/notification-dot/NotificationDot.tsx:13]: do we not have this dot already? just in mobile i guess?

**Review by dylanjeffers — COMMENTED** [review]
just adding some comments :)

File: packages/web/src/components/notification-dot/NotificationDot.tsx (hunk 3 above)
Comment (schottra) [reply packages/web/src/components/notification-dot/NotificationDot.tsx:13]: Just in mobile, correct. We had a sort of similar dot for when playlists are updated on desktop. But it was messy. This is formalizing it as a reusable component.

File: packages/web/src/components/Icon/Icon.tsx (hunk 1 above)
Comment (schottra) [reply packages/web/src/components/Icon/Icon.tsx:20]: Good callout I'll try it and see if it affects the positioning behavior.

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/pay-925-new-message-indicator

File: packages/web/src/components/Icon/Icon.tsx (hunk 2 above)
Comment (dylanjeffers) [reply packages/web/src/components/Icon/Icon.tsx:25]: interesting, i do believe it is the case that spreading doesn't lead to a unnecessary rerenders

Comment (schottra) [reply packages/web/src/components/Icon/Icon.tsx:25]: I did some investigating:

**Pull request merged**